x-pack/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/use_add_bulk_to_timeline.tsx

```diff
@@ -0,0 +1,230 @@
+
+export interface UseAddBulkToTimelineActionProps {
+  /* filters being passed to the Alert/events table */
+  localFilters: Filter[];
+  /* Table ID for which this hook is being used */
+  tableId: TableId;
+  /* start time being passed to the Events Table */
+  from: string;
+  /* End Time of the table being passed to the Events Table */
+  to: string;
+  /* Sourcerer Scope Id*/
+  scopeId: SourcererScopeName;
+}
+
+/*
+ * useAddBulkToTimelineAction  returns a bulk action that can be passed to the
+ * TGrid so that multiple items at a time can be added to the timeline.
+ *
+ * It also syncs the timerange passed to the TGrid to that of  timeline
+ *
+ * */
+export const useAddBulkToTimelineAction = ({
+  localFilters,
+  tableId,
+  from,
+  to,
+  scopeId,
+}: UseAddBulkToTimelineActionProps) => {
+  const [disableActionOnSelectAll, setDisabledActionOnSelectAll] = useState(false);
+
+  const {
+    browserFields,
+    dataViewId,
+    runtimeMappings,
+    indexPattern,
+    // important to get selectedPatterns from useSourcererDataView
+    // in order to include the exclude filters in the search that are not stored in the timeline
+    selectedPatterns,
+  } = useSourcererDataView(scopeId);
+  const dispatch = useDispatch();
+  const { uiSettings } = useKibana().services;
+
+  const { filters, dataTable: { selectAll, totalCount, sort, selectedEventIds } = tableDefaults } =
+    useSelector((state: State) => eventsViewerSelector(state, tableId));
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const esQueryConfig = useMemo(() => getEsQueryConfig(uiSettings as any), [uiSettings]);
+
+  const timelineQuerySortField = sort.map(({ columnId, columnType, esTypes, sortDirection }) => ({
+    field: columnId,
+    direction: sortDirection as Direction,
+    esTypes: esTypes ?? [],
+    type: columnType,
+  }));
+
+  const combinedFilters = useMemo(() => [...localFilters, ...filters], [localFilters, filters]);
+
+  const combinedQuery = combineQueries({
+    config: esQueryConfig,
+    dataProviders: [],
+    indexPattern,
+    filters: combinedFilters,
+    kqlQuery: { query: '', language: 'kuery' },
+    browserFields,
+    kqlMode: 'filter',
+  });
+
+  const filterQuery = useMemo(() => {
+    if (!combinedQuery) return '';
+    return combinedQuery.filterQuery;
+  }, [combinedQuery]);
+
+  // in case user selects all items, we will use below handler to get the IDs of
+  // all items (with max limit)
+  const [, , searchhandler] = useTimelineEventsHandler({
+    dataViewId,
+    endDate: to,
+    startDate: from,
+    id: tableId,
+    fields: ['_id', 'timestamp'],
+    sort: timelineQuerySortField,
+    indexNames: selectedPatterns,
+    filterQuery,
+    runtimeMappings,
+    limit: Math.min(MAX_LIMIT_TIMELINE, totalCount),
+    timerangeKind: 'absolute',
+  });
+
+  useEffect(() => {
+    if (!selectAll) {
+      setDisabledActionOnSelectAll(false);
+      return;
+    }
+    if (totalCount > 4000) {
```

Comment:
Should this now be `totalCount > MAX_LIMIT_TIMELINE` ?

Reply:
done.